usersession/userd/privileged_desktop_launcher.go

```diff
@@ -75,6 +79,13 @@ func (s *PrivilegedDesktopLauncher) IntrospectionData() string {
 // DBus interface. The desktopFileID is described here:
 // https://standards.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#desktop-file-id
 func (s *PrivilegedDesktopLauncher) OpenDesktopEntry(desktopFileID string, sender dbus.Sender) *dbus.Error {
+	return s.OpenDesktopEntryWithArguments(desktopFileID, nil, sender)
+}
+
+func (s *PrivilegedDesktopLauncher) OpenDesktopEntryWithArguments(desktopFileID string, arguments []string, sender dbus.Sender) *dbus.Error {
```

Comment:
As we are adding a new API here, I think it'd be good to add extra `action string` and `launchEnv map[string]string` arguments that we can use to pass in the startup notification environment. That way we won't need to add a third variant right away.

It's fine if we immediately error out if `action != "" || len(launchEnv) != 0` for now: we can add the real support for it later.